File: backend/Domain/Entities/House.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;

namespace Domain.Entities
{
    public class House : BaseEntity
    {
        [ForeignKey(nameof(ResidentialProperty))]
        public Guid ResidentialPropertyId { get; set; }
        public int NumberOfStories { get; set; }
        public double GarageSpace { get; set; }
        public bool StudentFriendly { get; set; }

        public ResidentialProperty ResidentialProperty { get; set; } = null!;
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/house.dart
import 'package:equatable/equatable.dart';

class House extends Equatable {
  final int numberOfStories;
  final double garageSpace;
  final bool studentFriendly;

  const House({
    required this.numberOfStories,
    required this.garageSpace,
    required this.studentFriendly,
  });

  @override
  List<Object?> get props => [numberOfStories, garageSpace, studentFriendly];
}
